xarray/core/computation.py

```diff
@@ -1917,7 +1917,7 @@ def polyval(
     return res
 
 
-def _ensure_numeric(data: T_Xarray) -> T_Xarray:
+def _ensure_numeric(data: Dataset | DataArray) -> Dataset | DataArray:
```

Comment:
This is the kind of func that would be nice at some point to make generic; with the proposed code we lose whether it's a `Dataset` vs. `DataArray`. (fine to add as a comment / TODO tho)

Reply:
I failed to make it work with Typevars since it is called with `DataArray|Dataset` :(